### ISSUE TITLE
Jest detects memory leak when requiring passport

### ISSUE THREAD

**Issue by jcasner** [jaredhanson/passport#711]
While tracking down some out of memory issues in our CI pipelines, I discovered that Jest's experimental `--detectLeaks` flag is detecting a memory leak in `passport` simply by requiring passport (`const passport = require('passport');`).

### Expected behavior
No memory leaks detected

### Actual behavior
Memory leak detected:
```
> jest --detectLeaks --config ./jest.config.js

 FAIL  src/index.spec.js
  ● Test suite failed to run

    EXPERIMENTAL FEATURE!
    Your test suite is leaking memory. Please ensure all references are cleaned.

    There is a number of things that can leak memory:
      - Async operations that have not finished (e.g. fs.readFile).
      - Timers not properly mocked (e.g. setInterval, setTimeout).
      - Keeping references to the global scope.

      at node_modules/jest/node_modules/jest-cli/build/TestScheduler.js:275:22

Test Suites: 1 failed, 1 total
```

### Steps to reproduce
I created a [minimal repo](https://github.com/jcasner/test-passport-jest-memory-leak) to reproduce this issue.
```
git clone git@github.com:jcasner/test-passport-jest-memory-leak.git
nvm use
npm i
npm run test
```

### Environment

* Operating System: MacOS 10.14.2 (Mojave)
* Node version: 10.4.1
* passport version: 0.4.0

**Comment by jaredhanson** [comment]
Could you do some investigation into what specific variables or references are triggering this warning?

**Comment by jcasner** [comment]
I will try to find some time to dig into the passport internals. If you check out the minimal reproduction set, there is only 1 test case and it fails with the following code:
```javascript
const passport = require('passport');

describe('tests', () => {
  it('does nothing', () => {
    expect(true).toBe(true);
  });
});
```
This suggests that if it's related to a variable or reference within passport that it's one of the defaults...

**Comment by Pasukaru** [comment]
Any updates on this?

**Comment by gbataille** [comment]
I have come across the same issue.

Investigating a bit (by commenting things out to see when `jest` complains and when it does not, the cause in the monkey patching of `http` in `lib/framework/connect.js`

But I don't think that's an actual issue, is it?

`--detectLeaks` seems to be very experimental. A lot of issues are opened on the `jest` repo.  
I also investigated something on the `faker` library that similarly trigger `jest` leak alarm, but it's unclear that there is indeed a leak associated

**Comment by jaredhanson** [comment]
This should be fixed in `passport@0.5.0`, by way of #848 which removes monkey patching of the core `http` module (which is the root cause of the memory leak - in that it added functions to global scope).

For more information, see the [announcement](https://medium.com/passportjs/passport-v0-5-0-cce2245e3d3c) on the blog and [details](https://github.com/jaredhanson/passport/blob/master/CHANGELOG.md#050---2021-09-23) in the CHANGELOG.
